pymeasure/instruments/newport/smc100.py

```diff
@@ -0,0 +1,706 @@
+        check_get_errors=True,
+        cast=str
+    )
+    state = Instrument.measurement(
+        "TS",
+        """Get the controller state.
+
+        Be careful: this command erases the last positioner error, which
+        should be checked first if it is of interest.""",
+        get_process=lambda v: ControllerState(v[-2:]),
+        check_get_errors=True,
+        cast=str
+    )
+    unit = Instrument.control(
+        "SU?", "SU%g",
+        """Get or set the value for one encoder count in the unit which
+        is thus defined.
+
+        Type: Floating point number
+        Range: ]1e-6, 1e12[
+        Unit: u""",
+        validator=strict_range,
+        values=(1e-6, 1e12),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+    velocity = Instrument.control(
+        "VA?", "VA%g",
+        """Get or set the velocity.
+
+        Type: Floating point number
+        Range: ]1e-6, 1e12[
+        Unit: u/s""",
+        validator=strict_range,
+        values=(1e-6, 1e12),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+    voltage = Instrument.control(
+        "DV?", "DV%g",
+        """Get or set the maximum driver voltage.
+
+        Type: Floating point number
+        Range: [12, 48]
+        Unit: V""",
+        validator=strict_range,
+        values=(12, 24),
+        check_get_errors=True,
+        check_set_errors=True
+    )
+
+    # Special methods
+    def __init__(self, adapter, address=1, name="SMC100", **kwargs):
+        """Set up connection to the controller."""
+        self.address = str(address)
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            baud_rate=57600,
+            flow_control=ControlFlow.xon_xoff,
+            preprocess_reply=lambda v, address=self.address: sub(
+                f"^{address}[A-Z]{{2}}[L,R,T]?", "", v),
+            **kwargs
+        )
+
+    def __str__(self):
+        """Return a string describing the controller settings."""
+        message = "Newport SMC100 Motion Controller with the following\n"
+        message += "settings (u stands for mm or deg by default):\n"
+        message += f"* address: {self.address}\n"
+        for k, v in self.__class__.__dict__.items():
+            if type(v) is property and v.fget.__defaults__[0] is not None:
+                unit = ""
+                try:
+                    unit += v.__doc__.split("Unit:")[1].split("\n")[0]
+                except IndexError:
+                    pass
+                message += f"* {k}: {v.fget(self)}{unit}\n"
+        return message[:-1]
+
+    # Implementation or extension of Instrument parent class methods
+    def check_errors(self):
+        """Check for command error except when broadcasting."""
+        if self.address != "0":
+            self.check_command_error()
+
+    def ask(self, command):
+        return super().ask(self.address + command)
```

Comment:
@bmoneke we should [not override](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#adding-a-device-address-and-adding-delay) `ask` anymore, nor `values`, only `write` to prepend the address, right?

Reply:
Exactly. I'll write a short, general note in the main thread.